src/vehicle_base.h

```diff
@@ -756,6 +791,7 @@ struct Vehicle : VehiclePool::PoolItem<&_vehicle_pool>, BaseVehicle, BaseConsist
 	void UpdateVisualEffect(bool allow_power_change = true);
 	void ShowVisualEffect() const;
 
+	void Travel(int x, int y);
 	void UpdatePosition();
```

Comment:
might be an idea to make this function protected/private to encourage Travel's usage?

Reply:
Looked at it, and it would be a ton of work to do this. Almost all the remaining uses are in vehicle creation functions (typically command handlers) that are outside the class scope. Making `UpdatePosition()` protected or private would mean adding 12 or so friend declarations, or doing a large amount of code restructuring.